src/core/cluster.py

```diff
@@ -213,7 +263,7 @@ def planned_units(self) -> int:
         return self.model.app.planned_units()
 
     @property
-    def ready_to_start(self) -> Status:
+    def ready_to_start(self) -> Status:  # noqa: C901
```

Comment:
Because of all the if statements, the cyclomatic complexity of this method is above our quality threshold. I silenced [this specific rule](https://docs.astral.sh/ruff/rules/#mccabe-c90) here, but we could rework the entire thing to not have any ifs. We could use something like [itertools.compress](https://docs.python.org/3/library/itertools.html#itertools.compress).

What do you think? I do not mind at all, but I was a little worried that this was a "too clever one-liner for our own good" solution